### ISSUE TITLE
Invoke multiple storage.stores in parallel

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-services/contile#338]
We could probably `join!` the potentially multiple calls to `StorageImage::store` to invoke them all in parallel. This may not be a huge win since there's currently a limit of 2 tiles and some of the work is CPU based (image hashing), but there's also at least one and potentially two http requests per store call.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/CONSVC-1499)
┆epic: Contile efficiency enhancements

**Comment by jrconlin** [comment]
Duplicate of #232
